extra-tests/classes/RollupFullRecalcTests.cls

```diff
@@ -2244,4 +2245,116 @@ private class RollupFullRecalcTests {
     acc = [SELECT AnnualRevenue FROM Account];
     System.assertEquals(opp.Amount, acc.AnnualRevenue);
   }
+
+  @IsTest
+  static void retainsMultiCurrencyFieldsOnParentRecords() {
+    if (RollupCurrencyInfo.isMultiCurrency() == false) {
+      return;
+    }
+    String expectedIsoCode = 'EUR';
+    Account acc = (Account) RollupTestUtils.queryRecord('Account', new List<String>());
+    System.assertNotEquals(null, acc.get(RollupCurrencyInfo.CURRENCY_ISO_CODE_FIELD_NAME), 'Test has started under the wrong conditions');
+    acc.put(RollupCurrencyInfo.CURRENCY_ISO_CODE_FIELD_NAME, expectedIsoCode);
+
+    RollupDeferredFullRecalcProcessor processor = new RollupDeferredFullRecalcProcessor(
+      new List<Rollup__mdt>{ new Rollup__mdt(RollupFieldOnLookupObject__c = 'Description', LookupObject__c = 'Account') },
+      Schema.Contact.SObjectType,
+      'SELECT Id, CurrencyIsoCode\nFROM Contact WHERE Id != null',
+      new Set<String>(),
+      Rollup.InvocationPoint.FROM_FULL_RECALC_LWC,
+      null
+    );
+
+    List<Account> storedAccounts = new List<Account>{ acc };
+    processor.storeParentFieldsToClear(storedAccounts);
+    List<Account> newAccounts = new List<Account>{ new Account(Id = RollupTestUtils.createId(Account.SObjectType)), acc };
+    newAccounts[0].put(RollupCurrencyInfo.CURRENCY_ISO_CODE_FIELD_NAME, 'USD');
+    processor.processParentFieldsToReset(newAccounts);
+
+    System.assertEquals(expectedIsoCode, acc.get(RollupCurrencyInfo.CURRENCY_ISO_CODE_FIELD_NAME), 'Currency ISO code should not have been cleared');
+    System.assertNotEquals(null, newAccounts[0].get(RollupCurrencyInfo.CURRENCY_ISO_CODE_FIELD_NAME));
+  }
+
+  @IsTest
+  static void correctlyHandlesMultipleDeferredFullRecalcs() {
+    // TODO continue on this thread for deferred full recalcs that are themselves INSIDE rollup conductors
+    Account acc = [SELECT Id FROM Account];
+    ContactPointAddress one = new ContactPointAddress(ParentId = acc.Id, Name = 'child one', PreferenceRank = 50);
```

Comment:
It seems highly unlikely to happen, but it might be worth using 2 different values on the `ContactPointAddress` records (like 25 & 75, instead of 50 & 50) just to ensure that the code is counting data from both records, and not somehow counting 1 record twice.

Reply:
great call out - I will definitely make this change!